_posts/2023-05-20-quarkus-3-upgrade.adoc

```diff
@@ -0,0 +1,95 @@
+---
+layout: post
+title: 'Migration to Quarkus 3.0 is a breeze'
+date: 2023-05-20
+tags: migration
+synopsis: 'Quarkus 3.0, our new major release, is out. Now is the time to migrate!'
+author: mkouba
+---
+
+Quarkus 3 is out and it's about the time to update your application!
+The team worked hard to deliver a lot of new features and performance improvements.
+But of course, there are also some breaking changes.
+The https://github.com/quarkusio/quarkus/wiki/Migration-Guide-3.0[migration guide] describes all the important changes in the core extensions.
+
+Nevertheless, there is also some tedious work that needs to be done.
+Quarkus 3 has updated to the latest Jakarta EE 10 APIs, which include the infamous change of the `javax.\*` to the `jakarta.*` packages.
+If your application makes use of any Jakarta API, and there's a good chance it does (think about JAX-RS, CDI, JPA, etc.), you'll need to modify your code.
+Furthermore, if there's an explicit Jakarta API dependency in the project you'll need to update this dependency as well.
+
+Of course, you can migrate the application manually.
+But since Quarkus embraces the developer joy there is the Quarkus Update Tool to the rescue!
+
+== Quarkus Update Tool - TL;DR
+
+. Install the https://quarkus.io/guides/cli-tooling[Quarkus CLI] or make sure a 3+ version is installed
+. Run `quarkus update --stream=3.0`
+. Check the changes made by the tool
+. Profit!
+
+== Quarkus Update Tool - The Full Story
+
+The goal of the update tool is to help developers to migrate an application to Quarkus 3 smoothly.
+The update process can be triggered from the Quarkus CLI, or directly via Maven and Gradle commands.
```

Comment:
```suggestion
The update process can be triggered from the Quarkus CLI, or directly via Maven or Gradle commands.
```

Reply:
I'm not sure about this one. There are now two `or`s in the sentence :shrug:

Reply:
How about the current wording?